### ISSUE TITLE
No multi-chart layout composition (dashboard grids of charts)

### ISSUE THREAD

**Issue by ryanoneill** [ryanoneill/envision#369]
## Summary

There is no way to compose multiple Chart (or other visualization) components into a grid layout within a single view. Each chart must be rendered independently, occupying the full available area. You cannot create a 2×2 grid of charts, or a side-by-side comparison, or a dashboard with loss curves next to accuracy curves next to a heatmap.

## Why this matters

Multi-panel dashboards are the norm in ML training monitoring:

- **TensorBoard:** Multiple scalars, images, histograms in a scrollable grid
- **Weights & Biases:** Customizable panel grids with drag-and-drop arrangement
- **Plotly subplots:** `make_subplots(rows=2, cols=2)` is standard
- **matplotlib:** `fig, axes = plt.subplots(2, 2)` is day-one usage

A training dashboard typically shows 4-8 panels simultaneously:
- Loss curves (train + val)
- Primary metric (accuracy, F1, etc.)
- Learning rate schedule
- GPU utilization / throughput
- Gradient norms
- Weight distributions

Without layout composition, users must either:
- Render one chart at a time (losing the simultaneous overview)
- Build their own layout logic outside Envision (defeating the purpose of a framework)

MetricsDashboard provides a grid for simple metric widgets, but it cannot embed Chart, Heatmap, Histogram, or BoxPlot components — only its own Counter/Gauge/Status/Progress widget types.

## Suggested approach

A layout component that arranges arbitrary Envision components in a grid:

```rust
// Conceptual API
let dashboard = DashboardLayout::new(2, 2)  // 2 rows, 2 cols
    .set_panel(0, 0, ChartPanel::new(&loss_chart_state))
    .set_panel(0, 1, ChartPanel::new(&accuracy_chart_state))
    .set_panel(1, 0, HeatmapPanel::new(&confusion_state))
    .set_panel(1, 1, ChartPanel::new(&lr_chart_state));
```

This is architecturally significant — it requires either:
1. A trait-object approach where panels implement a common `Renderable` trait
2. An enum-based approach with known panel types
3. A closure-based approach where each cell gets a render callback

The simpler version is just providing a layout helper that splits an area into a grid of `Rect`s and lets the caller render into each cell. This doesn't require any architectural changes to Envision's component system:

```rust
let grid = GridLayout::new(2, 2).split(area);
Chart::view(&loss_state, frame, grid[0][0], &theme, &ctx);
Chart::view(&acc_state, frame, grid[0][1], &theme, &ctx);
Heatmap::view(&conf_state, frame, grid[1][0], &theme, &ctx);
Chart::view(&lr_state, frame, grid[1][1], &theme, &ctx);
```
